### PR TITLE
EXUI-1108: Update to yarn v4.5.0

### PR DESCRIPTION
**Pull request by Josh-HMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EXUI-1108

### Change description ###

Update yarn to v4.5.0, update yarn audit sh script to work with new yarn audit format

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by RiteshHMCTS — APPROVED** [review]

**Review by connorpgpmcelroy — DISMISSED** [review]
LGTM

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Good effort but let's discuss possible changes

File: bin/run-yarn-audit.sh [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+fi
 
-    fi
+if [[ ! -f "$vulnerabilitiesInRepo" ]]; then
+  echo "File $vulnerabilitiesInRepo does not exist."
+  exit 1
 fi
+
+# Compare the files and output the differences
+diff_output=$(diff "$upToDateVulnerabilities" "$vulnerabilitiesInRepo")
```
[the remark this answers is not in the document]
Comment (Josh-HMCTS) [reply bin/run-yarn-audit.sh:22]: Yeah good point, i will update this to check the return code and not store the diff output

File: bin/run-yarn-audit.sh [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
+
+if [[ -n "$diff_output" ]]; then
```
[the remark this answers is not in the document]
Comment (Josh-HMCTS) [reply bin/run-yarn-audit.sh:24]: I'll update the -n to just use the return code as part of above change
Ill just remove the diff output from the print as i dont think it really adds anything as anyone who sees its failed will just run the command again not use the diff output to manually update it

**Review by olusegz07 — APPROVED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**